contrib/segyconverter/utils/segyextract.py

```diff
@@ -0,0 +1,332 @@
+    start with. SEGY files can be non standard and use other byte location for
+    these values. In that case, the data from this method will be erroneous. It
+    is up to the user to figure out which numbers to use by reading the SEGY text
+    header and finding the byte offsets visually.
+
+    :param str input_file: path to segy file
+    :param int iline: inline byte position
+    :param int xline: crossline byte position
+    :returns: fast_distinct, slow_distinct, trace_headers, samplesize
+    :rtype: list, DataFrame, DataFrame, int
+    """
+
+    with segyio.open(input_file, ignore_geometry=True) as segy_file:
+        segy_file.mmap()
+        # Initialize df with trace id as index and headers as columns
+        trace_headers = pd.DataFrame(index=range(0, segy_file.tracecount),
+                                     columns=['i', 'j'])
+
+        # Fill dataframe with all trace headers values
+        trace_headers['i'] = segy_file.attributes(iline)
+        trace_headers['j'] = segy_file.attributes(xline)
+
+        _identify_fast_direction(trace_headers, FAST, SLOW)
+
+        samplesize = len(segy_file.samples)
+        fast_distinct = _remove_duplicates(trace_headers[FAST])
+        slow_distinct = np.unique(trace_headers[SLOW])
+    return fast_distinct, slow_distinct, trace_headers, samplesize
+
+
+def process_segy_data_into_single_array(input_file, output_dir, prefix, iline=189, xline=193):
+    """
+    Open segy file and write all data to single npy array
+    :param str input_file: path to segyfile
+    :param str output_dir: path to directory where npy files will be written/rewritten
+    :param str prefix: prefix to use when writing npy files
+    :param int iline: iline header byte location
+    :param int xline: crossline header byte location
+    :returns: 3 dimentional numpy array of SEGY data
+    :rtype: nparray
+    """
+    fast_distinct, slow_distinct, trace_headers, sampledepth = get_segy_metadata(input_file, iline, xline)
+    with segyio.open(input_file, ignore_geometry=True) as segy_file:
+        segy_file.mmap()
+
+        fast_line_space = abs(fast_distinct[1] - fast_distinct[0])
+
+        slow_line_space = abs(slow_distinct[0] - slow_distinct[1])
+        sample_size = len(segy_file.samples)
+        layer_fastmax = max(fast_distinct)
+        layer_fastmin = min(fast_distinct)
+        layer_slowmax = max(slow_distinct)
+        layer_slowmin = min(slow_distinct)
+        layer_trace_ids = trace_headers[(trace_headers.fast >= layer_fastmin)
+                                        & (trace_headers.fast <= layer_fastmax)
+                                        & (trace_headers.slow >= layer_slowmin)
+                                        & (trace_headers.slow <= layer_slowmax)]
+
+        block = np.zeros((len(fast_distinct), len(slow_distinct), sampledepth), dtype=np.float32)
+        for _, row in layer_trace_ids.iterrows():
+            block[(row[FAST] - layer_fastmin) // fast_line_space,
```

Comment:
I see a lot of slow and fast in the code - can these be carried into a separate YACS config? so row[config.FAST] then names can be changed.